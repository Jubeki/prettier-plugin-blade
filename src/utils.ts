import { format } from "prettier";
// @ts-ignore
import php from "@prettier/plugin-php/standalone";

export const formatAsPhp = (source: string): string => {
  if (!source.startsWith("<?php")) {
    source = "<?php " + source;
  }

  let code = format(source, { parser: "php", plugins: [php] })
    .replace("<?php ", "")
    .trim();

  if (source.trim().endsWith(";")) {
    return code;
  }

  // The PHP plugin for Prettier will add a semi-colon by default. We don't always want that.
  return code.substring(0, code.length - 1);
};
