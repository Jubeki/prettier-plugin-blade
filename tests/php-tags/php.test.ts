import path from "path";
import { formatFile } from "../utils";

it("should convert php long tags to blade directives", function() {
  const formatted = formatFile(path.join(__dirname, "./php-long-tags.blade.php"));
  expect(formatted).toMatchSnapshot();
});

it("should convert php short tags to blade directives", function() {
  const formatted = formatFile(path.join(__dirname, "./php-short-tags.blade.php"));
  expect(formatted).toMatchSnapshot();
});

it("should convert php echo tags to raw echo", function() {
  const formatted = formatFile(path.join(__dirname, "./php-echo-tags.blade.php"));
  expect(formatted).toMatchSnapshot();
});