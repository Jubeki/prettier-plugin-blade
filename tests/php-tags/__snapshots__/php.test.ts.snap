// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should convert php echo tags to echo 1`] = `
"{{!! \\"Cool stuff\\" !!}}
<h1>Awesome</h1>
"
`;

exports[`should convert php long tags to blade directives 1`] = `
"@php
    echo \\"long way there\\";
@endphp
<h1>Awesome</h1>
"
`;

exports[`should convert php short tags to blade directives 1`] = `
"@php
    echo \\"Some to be happy about\\";
@endphp
<h1>Awesome</h1>
"
`;
